### ISSUE TITLE
Deleted releases

### ISSUE THREAD

**Issue by mans0954** [hashicorp/terraform-provider-helm#54]
If a release is provisioned with Terraform via this provider and then manually deleted with the command
```
helm delete <name>
```
then running `terraform apply` does not recreate the release. This is because Helm tracks releases even after they've been deleted. The provider will update the release metadata, running
```
terraform state show helm_release.<name>
```
will show that `metadata.0.status` is now `DELETED`. I would have expected the provider to attempt to rollback or recreate the release.

A work around is to purge the release from helm before applying:
```
helm delete <name> --purge
terraform apply
```

Thanks.

**Comment by gregdhill** [comment]
The problem with this provider is that it never refreshes it's internal state with Tiller, see: https://medium.com/monaxhq/evolving-infrastructure-881bbddd2ff9

We actually ended up writing our own pipelining tool called [Compass](https://github.com/gregdhill/compass) which effectively does what this does without the extra overhead.

**Comment by legal90** [comment]
That issue should be solved by https://github.com/terraform-providers/terraform-provider-helm/pull/161, but the fix is only in the master branch at this moment (e.q. it's not included to `v0.7.0`)

**Comment by mcuadros** [comment]
Closing this issue since is making reference to a version based on Helm 2, if this is still valid to the master branch please reopen it. Thanks.
